lib/handlers/htlc_swaps_handler.dart

```diff
@@ -0,0 +1,274 @@
+          P2pSwapState.reclaimable
+        ]);
+        if (unresolvedSwaps.isNotEmpty) {
+          if (await _areThereNewHtlcBlocks()) {
+            final newBlocks = await _getNewHtlcBlocks(unresolvedSwaps);
+            await _goThroughHtlcBlocks(newBlocks);
+          }
+          await _checkForExpiredSwaps();
+          _checkForAutoUnlockableSwaps();
+        }
+        sl<AutoUnlockHtlcWorker>().autoUnlock();
+      }
+    } catch (e) {
+      Logger('HtlcSwapsHandler').log(Level.WARNING, '_runPeriodically', e);
+    } finally {
+      Future.delayed(const Duration(seconds: 5), () => _runPeriodically());
+    }
+  }
+
+  Future<int?> _getHtlcFrontierHeight() async {
+    try {
+      final frontier = await zenon!.ledger.getFrontierAccountBlock(htlcAddress);
+      return frontier?.height;
+    } catch (e, stackTrace) {
+      Logger('HtlcSwapsHandler')
+          .log(Level.WARNING, '_getHtlcFrontierHeight', e, stackTrace);
+    }
+    return null;
+  }
+
+  Future<bool> _areThereNewHtlcBlocks() async {
+    final frontier = await _getHtlcFrontierHeight();
+    return frontier != null &&
+        frontier > htlcSwapsService!.getLastCheckedHtlcBlockHeight();
+  }
+
+  Future<List<AccountBlock>> _getNewHtlcBlocks(List<HtlcSwap> swaps) async {
+    final lastCheckedHeight = htlcSwapsService!.getLastCheckedHtlcBlockHeight();
+    final oldestSwapStartTime = _getOldestSwapStartTime(swaps) ?? 0;
+    int lastCheckedBlockTime = 0;
+
+    if (lastCheckedHeight > 0) {
+      try {
+        lastCheckedBlockTime =
+            (await AccountBlockUtils.getTimeForAccountBlockHeight(
+                    htlcAddress, lastCheckedHeight)) ??
+                lastCheckedBlockTime;
+      } catch (e, stackTrace) {
+        Logger('HtlcSwapsHandler')
+            .log(Level.WARNING, '_getNewHtlcBlocks', e, stackTrace);
+        return [];
+      }
+    }
+
+    try {
+      return AccountBlockUtils.getAccountBlocksAfterTime(
+          htlcAddress, max(oldestSwapStartTime, lastCheckedBlockTime));
+    } catch (e, stackTrace) {
+      Logger('HtlcSwapsHandler')
+          .log(Level.WARNING, '_getNewHtlcBlocks', e, stackTrace);
+      return [];
+    }
+  }
+
+  Future<void> _goThroughHtlcBlocks(List<AccountBlock> blocks) async {
+    for (final block in blocks) {
+      await _extractSwapDataFromBlock(block);
+      await htlcSwapsService!.storeLastCheckedHtlcBlockHeight(block.height);
+    }
+  }
+
+  Future<void> _extractSwapDataFromBlock(AccountBlock htlcBlock) async {
+    if (htlcBlock.blockType != BlockTypeEnum.contractReceive.index) {
+      return;
+    }
+
+    final pairedBlock = htlcBlock.pairedAccountBlock!;
+    final blockData = AccountBlockUtils.getDecodedBlockData(
+        Definitions.htlc, pairedBlock.data);
+
+    if (blockData == null) {
+      return;
+    }
+
+    final swap = _tryGetSwapFromBlockData(blockData);
+    if (swap == null) {
+      return;
+    }
+
+    if (swap.chainId != pairedBlock.chainIdentifier) {
+      return;
```

Comment:
Could this potentially impact users in the future if we have more than one mainnet chain identifier?

Example: I've been using a ZNN L2 node w/ chain id 2, wallet is manually set to chain id 2.
I connect to an L1 node (chain id 1) where I have some HTLCs waiting for me, but I don't set the wallet's chain id back to 1 in time.

Reply:
This should not be a problem since the handler only fetches the swaps from the DB that have been made on the connected node's chain ID. The P2P swap implementation does not rely on the chain ID set by the user.